plugins/inputs/intel_powerstat/intel_powerstat.go

```diff
@@ -1,915 +1,1218 @@
-	packageCurrentDramPowerConsumption = "current_dram_power_consumption"
-	packageThermalDesignPower          = "thermal_design_power"
-	packageTurboLimit                  = "max_turbo_frequency"
-	packageUncoreFrequency             = "uncore_frequency"
-	packageCPUBaseFrequency            = "cpu_base_frequency"
-	percentageMultiplier               = 100
-)
-
-// PowerStat plugin enables monitoring of platform metrics (power, TDP) and Core metrics like temperature, power and utilization.
+// PowerStat plugin enables monitoring of platform metrics.
 type PowerStat struct {
-	CPUMetrics     []string        `toml:"cpu_metrics"`
-	PackageMetrics []string        `toml:"package_metrics"`
-	Log            telegraf.Logger `toml:"-"`
-
-	fs   fileService
-	rapl raplService
-	msr  msrService
-
-	cpuFrequency        bool
-	cpuBusyFrequency    bool
-	cpuTemperature      bool
-	cpuC0StateResidency bool
-	cpuC1StateResidency bool
-	cpuC6StateResidency bool
-	cpuBusyCycles       bool
-
-	packageTurboLimit                  bool
-	packageCurrentPowerConsumption     bool
-	packageCurrentDramPowerConsumption bool
-	packageThermalDesignPower          bool
-	packageUncoreFrequency             bool
-	packageCPUBaseFrequency            bool
-
-	cpuBusClockValue   float64
-	cpuInfo            map[string]*cpuInfo
-	skipFirstIteration bool
-	logOnce            map[string]error
+	CPUMetrics       []string        `toml:"cpu_metrics"`
+	PackageMetrics   []string        `toml:"package_metrics"`
+	IncludedCPUs     []string        `toml:"included_cpus"`
+	ExcludedCPUs     []string        `toml:"excluded_cpus"`
+	EventDefinitions string          `toml:"event_definitions"`
+	MsrReadTimeout   config.Duration `toml:"msr_read_timeout"`
+	Log              telegraf.Logger `toml:"-"`
+
+	parsedIncludedCores []int
+	parsedExcludedCores []int
+
+	parsedCPUTimedMsrMetrics []string
+	parsedCPUPerfMetrics     []string
+	parsedPackageRaplMetrics []string
+	parsedPackageMsrMetrics  []string
+
+	option  OptGenerator
+	fetcher MetricFetcher
+
+	cpuMetricsMap    map[string]func(telegraf.Accumulator, int, int, int)
+	packageMetricMap map[string]func(telegraf.Accumulator, int)
+
+	needsCoreFreq       bool
+	needsMsrCPU         bool
+	needsPerf           bool
+	needsTimeRelatedMsr bool
+
+	needsRapl       bool
+	needsMsrPackage bool
+
+	logOnce map[string]struct{}
 }
 
+// SampleConfig returns a sample configuration (See sample.conf).
 func (*PowerStat) SampleConfig() string {
 	return sampleConfig
 }
 
-// Init performs one time setup of the plugin
-func (p *PowerStat) Init() error {
-	p.parsePackageMetricsConfig()
-	p.parseCPUMetricsConfig()
-	err := p.verifyProcessor()
-	if err != nil {
+// Start initializes the MetricFetcher interface of the receiver to gather metrics.
+func (p *PowerStat) Start(_ telegraf.Accumulator) error {
+	if err := p.disableUnsupportedMetrics(); err != nil {
+		return err
+	}
+
+	if err := p.parseConfig(); err != nil {
 		return err
 	}
```

Comment:
Parsing and validating telegraf config should be done in `Init()`

Reply:
Right I missed it in documentation. Done as part of commit [e2de6f6b](https://github.com/influxdata/telegraf/pull/14363/commits/e2de6f6b2577fe4e9ec7b420132b2750fe7e2654).